### PR TITLE
Fix invalid error messages if js_challenge points to invalid file

### LINKED ISSUE #1442: JSCH Template Issue
## Scope

With configuration:
```
listen 192.168.100.4:443 proto=https;
listen 192.168.100.4:80;

frang_limits {
	# Conenction level limits.
	client_header_timeout 20;
	client_body_timeout 10;
	http_header_chunk_cnt 10;
	http_body_chunk_cnt 0;

	# Message level limits (per-vhost)
	http_methods GET;
	http_uri_len 512;
	http_resp_code_block 400 403 404 3 10;
}

# Allow only following characters in URI (no '%'): /a-zA-Z0-9&?:-._=
http_uri_brange 0x2f 0x41-0x5a 0x61-0x7a 0x30-0x39 0x26 0x3f 0x3a 0x2d 0x2e 0x5f 0x3d;

sticky {
	cookie name=my_js_cookie enforce max_misses=3 timeout=3;
	js_challenge resp_code=503 delay_min=1000 delay_range=1000 delay_limit=3000 /root/tempesta/etc/js_challenge.html;
}

block_action attack reply;
block_action error reply;

srv_group default {
	server 127.0.0.1:8080 conns_n=4;
}

#vhost debian {
vhost default {
	tls_certificate /root/tempesta/etc/tfw-root.crt;
	tls_certificate_key /root/tempesta/etc/tfw-root.key;

	resp_hdr_set Strict-Transport-Security "max-age=31536000; includeSubDomains";

	proxy_pass default;
}

cache 0;
#cache_fulfill * *;

http_chain {
	#hdr "Referer" == "http://badhost.com*" -> block;

	#-> debian;
	-> default;
}
```

I get perl error:
```
Starting Tempesta...
...enable RPS on enp3s0
...load Tempesta modules
Loading Tempesta kernel modules...
Loading module tempesta_lib 
Loading module tempesta_tls 
Loading module tempesta_db 
Loading module tempesta_fw tfw_cfg_path=/home/user/qtc/last-build/etc/tempesta_fw.conf
...compile html templates for JS challenge
Use of uninitialized value $src in -e at /home/user/qtc/last-build/scripts/update_template.pl line 37.
Use of uninitialized value $src in concatenation (.) or string at /home/user/qtc/last-build/scripts/update_template.pl line 38.
JavaScript Challenge template '' not found, generation aborted! at /home/user/qtc/last-build/scripts/update_template.pl line 38.
...start Tempesta FW
Un-loading Tempesta kernel modules...
ERROR: cannot start Tempesta FW (sysctl code: Invalid argument)
```

I have no file `/root/tempesta/etc/js_challenge*` on my VM.

## Testing

Check what cause the problem add add the test to JSCH functional tests.

### PR DESCRIPTION
**Pull request by ikoveshnikov**
fix #1442

### CODE REVIEW

File: scripts/update_template.pl [hunk 1]
```diff
@@ -35,7 +35,7 @@ sub assemble
 {
 	my ($src) = @_;
 	unless (-e $src) {
-		die "JavaScript Challenge template '$src' not found, generation aborted!";
+		die "JavaScript Challenge template '$src' not found, generation aborted!\n";
```
Comment (krizhanovsky) [inline scripts/update_template.pl:38]: `die` adds line number and LF, so no need for `\n`

File: scripts/update_template.pl [hunk 2]
```diff
@@ -58,6 +58,9 @@ sub assemble
 		|| die $t->error(), "\n";
 }
 
+unless (-e $template && -r $template) {
+	die "JavaScript Challenge template '$template' not found, generation aborted!\n";
+}
```
Comment (krizhanovsky) [inline scripts/update_template.pl:63]: I think it's better to move the check higher, where we check all the arguments from `@ARGV`

**Review by krizhanovsky — APPROVED** [review]
LGTM

File: scripts/update_template.pl (hunk 1 above)
Comment (ikoveshnikov) [reply scripts/update_template.pl:38]: `die` adds not only LF, but also a trace to faulty file and line. This information is omitted if explicit `\n` was stated. For user the trace message looks like internal error in our scripts, but it actually configuration error.

**Pull request merged**